ports/esp32/machine_timer.c

```diff
@@ -238,6 +246,7 @@ static mp_obj_t machine_timer_init_helper(machine_timer_obj_t *self, mp_uint_t n
     self->repeat = args[ARG_mode].u_int;
     self->handler = machine_timer_isr_handler;
     self->callback = args[ARG_callback].u_obj;
+    self->ishard = args[ARG_hard].u_bool;
```

Comment:
It might be just me but I read `ishard` as "I shard", personally I'd go for `is_hard` or just `hard`. Minor detail though...